pulp_ansible/app/tasks/collections.py

```diff
@@ -431,6 +430,22 @@ async def _add_collection_version(self, api_version, collection_version_url, met
 
         info = metadata["metadata"]
 
+        if self.add_dependents:
+            dependencies = info["dependencies"]
+            tasks = []
+            loop = asyncio.get_event_loop()
+            for name in dependencies.keys():
+                # get everything to avoid adding potential duplicates to pipeline
+                new_req = RequirementsFileEntry(
+                    name=name,
+                    version="*",
```

Comment:
I think we want to use the version specifier declared by the dependency itself. If none is specified we should use `*` in that case.

Reply:
Did you see my comment above? https://github.com/pulp/pulp_ansible/pull/545#discussion_r597069389 

There is an interesting scenario where say collection C has 10 versions and collection A has a dependency on collection C version <= 9.0 and collection B has a dependency on collection C any version.  If collection A is first in the requirements file it'll see the dependency and add the 9 versions of C to the sync pipeline.  Then B will be synced and it'll see the dependency and since it's any version it'll add all 10 versions of C to the pipeline even though 9 versions are already in it.  Pulp will sort this out later on, but the sync has gotten slower and more api requests where made then needed.  Now if there weren't shared dependencies across collections your suggestion would reduce the amount of CVs added to the pipeline, but I think it's hard to tell what case is more common.  I'm fine doing it either way, but I wanted to share my reasoning for having it like this.

Reply:
If that's how it's working in that example, I think we may do better having it deduplicate just before actually sending specific collection versions down the pipeline. I think this would happen in `_add_collection_version` metadata itself. Having a dictionary that stors the name, namespace, version triplet that was sync'd and ensuring that isn't already present I think would do it. What do you think about this alternative?